File: src/leetcode/python/medium/LargestNumber_179.py
class LargestNumber_179(object):
    # @param {integer[]} nums
    # @return {string}
    def largestNumber(self, nums):
        r = ''.join(sorted(map(str, nums), lambda a,b: [1, -1][a+b>b+a]))
        return r.lstrip('0') or '0'


l = LargestNumber_179()
a = [1, 2, 3, 4]
print(l.largestNumber(a))
print(str(a))
print(''.join(str(i) for i in a))

File: src/leetcode/java/medium/LargestNumber_179.java
package leetcode.java.medium;

import java.util.Arrays;

public class LargestNumber_179 {
  public String largestNumber(int[] nums) {
    if (nums == null || nums.length == 0) return "";
    String[] strs = new String[nums.length];
    for (int i = 0; i < nums.length; i++)
      strs[i] = nums[i] + "";
    Arrays.sort(strs, (a, b) -> ((a + b).compareTo(b + a))); 
    StringBuilder builder = new StringBuilder();
    for (int i = strs.length - 1; i >= 0; i--)
      builder.append(strs[i]);
    String res = builder.toString();
    if (res.charAt(0) == '0') return "0";
    return res;
  }
}
